src/operator/mergeMapTo.ts

```diff
@@ -73,17 +71,27 @@ export class MergeMapToSubscriber<T, R, R2> extends OuterSubscriber<T, R> {
              innerSub: InnerSubscriber<T, R>): void {
     const { resultSelector, destination } = this;
     if (resultSelector) {
-      const result = tryCatch(resultSelector)(outerValue, innerValue, outerIndex, innerIndex);
-      if (result === errorObject) {
-        destination.error(errorObject.e);
-      } else {
-        destination.next(result);
-      }
+      this.trySelectResult(outerValue, innerValue, outerIndex, innerIndex);
     } else {
       destination.next(innerValue);
     }
   }
 
+  private trySelectResult(outerValue: T, innerValue: R,
+                          outerIndex: number, innerIndex: number): void {
+    const { resultSelector, destination } = this;
+    let result: R2;
+    try {
+      result = resultSelector(outerValue, innerValue, outerIndex, innerIndex);
+    } catch (err) {
+      destination.error(err);
+    }
+
+    if (result) {
```

Comment:
I don't think there's a need to assert `if (result)` here. The resultSelector could _want_ to return `false` or `0`. So we can probably remove this.